### ISSUE TITLE
cmd/compile: better error message when generics are used with -lang=go1.17

### ISSUE THREAD

**Issue by mdempsky** [golang/go#47818]
Currently, when compiling with `-G=3 -lang=go1.17`, we run the types2 typechecker, but parse without the AllowGenerics flag. The result is when users try to write generic code, they'll get syntax error messages like `unexpected [, expecting (`, which seems to be a recurring source of confusion and leaves users (incl. expert Go users) uncertain of how to proceed.

I think we should probably instead always parse with AllowGenerics (at least when we're going to use types2), and have types2 report the usual `allowVersion` error messages if users try to use generics with go1.17 or earlier.

/cc @griesemer @findleyr

**Comment by griesemer** [comment]
Agreed. This makes sense now that we're on 1.18. I will take care of this (but I'm ooo until next Monday).
